monitor.cpp

```diff
@@ -4,35 +4,66 @@
 #include <chrono>
 #include <iostream>
 using std::cout, std::flush, std::this_thread::sleep_for, std::chrono::seconds;
-
+/// @brief displayWarning function to display warning message
+void displayWarning() {
+  for (int i = 0; i < 6; i++) {
+    cout << "\r* " << flush;
+    sleep_for(seconds(1));
+    cout << "\r *" << flush;
+    sleep_for(seconds(1));
+  }
+}
+/// @brief to check if the vitals are ok
+/// @param temperature
+/// @param pulseRate
+/// @param spo2
+/// @return 1 if vitals are ok, 0 otherwise
 int vitalsOk(float temperature, float pulseRate, float spo2) {
+  if (checkVitals(temperature, pulseRate, spo2)) {
+    return 1;
+  }
+  return 0;
+}
+
+bool checkVitals(float temperature, float pulseRate, float spo2) {
+  return (checkTemperature(temperature) && checkPulseRate(pulseRate) && checkSpo2(spo2));
+}
+/// @brief to check if the temperature is in range
+/// @param temperature
+/// @return true if temperature is in range, false otherwise
+bool checkTemperature(float temperature) {
   if (temperature > 102 || temperature < 95) {
     cout << "Temperature is critical!\n";
-    for (int i = 0; i < 6; i++) {
-      cout << "\r* " << flush;
-      sleep_for(seconds(1));
-      cout << "\r *" << flush;
-      sleep_for(seconds(1));
-    }
-    return 0;
-  } else if (pulseRate < 60 || pulseRate > 100) {
+    displayWarning();
+    return false;
+  }
+  return true;
+}
+/**
+ * @brief Checks if the pulse rate is within the acceptable range.
+ * 
+ * @param pulseRate The pulse rate value to be checked.
+ * @return True if the pulse rate is within the acceptable range, false otherwise.
+ */
+bool checkPulseRate(float pulseRate) {
+  if (pulseRate < 60 || pulseRate > 100) {
```

Comment:
Can further abstract the range checking into a function like `isInRange(float inputValue, float lowerLimit, float upperLimit)`
If you test that function, you don't need to add 3 asserts for covering every additional parameter added.